hilbert/factories.py

```diff
@@ -88,18 +141,29 @@ def build_mle_sample_solver(
         temperature=2,            # MLE option
         batch_size=10000,
         balanced=False,
```

Comment:
Please add one thing, make balanced default to `True`.
```suggestion
         balanced=True,
```